src/observable/GenerateObservable.ts

```diff
@@ -245,7 +245,7 @@ export class GenerateObservable<T, S> extends Observable<T> {
     } while (true);
   }
 
-  private static dispatch<T, S>(state: SchedulerState<T, S>) {
+  private static dispatch<T, S>(state: SchedulerState<T, S>): Subscription | void {
```

Comment:
can we consider this is returning `tearDownLogic`? it's bit broader than `subscription | void` though.